src/fty_alert_engine_server.cc

```diff
@@ -478,7 +485,17 @@ fty_alert_engine_stream(zsock_t *pipe, void* args) {
 
     zsock_signal(pipe, 0);
 
+    int64_t timeCash = zclock_mono();
+
+
     while (!zsys_interrupted) {
+
+        //clear cache every 1 min
+        if (zclock_mono() - timeCash > 30000) {
+            cache.removeOldMetrics();
+            timeCash = zclock_mono();
+        }
+
```

Comment:
Comment says 1 minute while the code is waiting 30s. But this kind of forced caching is really trading correctness for performance. It would make sense had we no control over the other agents, but we do have control over them. So if the performance problem is caused by noise on the METRICS stream (I suppose?), we should primarily address the noise. Arbitrarily ignoring metric updates in fty-alert-engine should be a last resort.

Reply:
I get it for the comment, but not the other point. Before the changes (and i apologies for the reformatting code) the performance problem was that when fty-alert-engine receive one metrics, he : 
-Call the removeOldMetrics which scan all metrics to remove any metrics with a ttl expired 
-Scan all rules to see if the metrics match with a rules and so evaluate it. For example an epdu have ~150 metrics, but  only ~20 metrics are evaluated.